python_examples/rebound.py

```diff
@@ -127,4 +149,158 @@ def step():
+        if inc != 0.: print('Omega not passed, setting to 0... (ascending node along x axis)')
+    if f != None and M != None: raise ValueError('Can only pass EITHER the true (f) or the mean (M) anomaly')
+    if f == None and M == None:
+        f = 0.;
+        if e != 0.:
+            print('Neither f nor M passed, setting initial angle from pericenter to 0...')
+        else:
+            print('Neither f nor M passed, setting particle on x axis...')
+    if mass == None: mass = 0.; print('mass not passed, setting to 0...')
+    
+    if not(0. <= e < 1.): raise ValueError('e must be in range [0,1)') # not sure if these equations work for parabolic/hyperbolic obits
+    if not(0. <= inc <= numpy.pi): raise ValueError('inc must be in range [0,pi]')
+    
+    if f == None: # need to calculate f
+        E = get_E(e,M)
+        f = mod2pi(2.*numpy.arctan(numpy.sqrt((1. + e)/(1. - e))*numpy.tan(0.5*E)))
+    
+    r = a*(1-e**2)/(1 + e*numpy.cos(f))
+    
+    # Murray & Dermott Eq. 2.122
+    _x  = star.x + r*(numpy.cos(Omega)*numpy.cos(omega+f) - numpy.sin(Omega)*numpy.sin(omega+f)*numpy.cos(inc))
+    _y  = star.y + r*(numpy.sin(Omega)*numpy.cos(omega+f) + numpy.cos(Omega)*numpy.sin(omega+f)*numpy.cos(inc))
+    _z  = star.z + r*numpy.sin(omega+f)*numpy.sin(inc)
+    
+    n = numpy.sqrt(get_G()*(star.m+mass)/(a**3))
+    
+    # Murray & Dermott Eq. 2.36 after applying the 3 rotation matrices from Sec. 2.8 to the velocities in the orbital plane
+    _vx = star.vx + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*(-numpy.cos(inc)*numpy.cos(omega)*numpy.sin(Omega) - numpy.cos(Omega)*numpy.sin(omega)) - numpy.sin(f)*(numpy.cos(omega)*numpy.cos(Omega) - numpy.cos(inc)*numpy.sin(omega)*numpy.sin(Omega)))
+    _vy = star.vy + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*(numpy.cos(inc)*numpy.cos(omega)*numpy.cos(Omega) - numpy.sin(omega)*numpy.sin(Omega)) - numpy.sin(f)*(numpy.cos(omega)*numpy.sin(Omega) + numpy.cos(inc)*numpy.cos(Omega)*numpy.sin(omega)))
+    _vz = star.vz + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*numpy.cos(omega)*numpy.sin(inc) - numpy.sin(f)*numpy.sin(inc)*numpy.sin(omega))
+    
+    return Particle(m  = mass, x = _x, y = _y, z = _z, vx = _vx, vy = _vy, vz = _vz)
+
+def p2orbit(p, star):
+    '''Takes 2 Particle objects (one for object and one for central body,
+        which hold Cartesian elements) and returns a Orbit object
+        with all the orbital elements'''
+    
+    if star.m <= 1.e-30: raise ValueError('Star has no mass.')
+    o = Orbit()
+    mu = get_G()*(p.m+star.m)
+    p.x -= star.x
+    p.y -= star.y
+    p.z -= star.z
+    p.vx -= star.vx
+    p.vy -= star.vy
+    p.vz -= star.vz
+    h0 = (p.y*p.vz - p.z*p.vy)                  # angular momentum vector
+    h1 = (p.z*p.vx - p.x*p.vz)
+    h2 = (p.x*p.vy - p.y*p.vx)
+    o.h = numpy.sqrt ( h0*h0 + h1*h1 + h2*h2 )  # abs value of angular moment
+    if o.h <= 1.e-30: raise ValueError('Particle orbit is radial.')
```

Comment:
Similar issue here, it's very good to have a failsafe in here, but it should probably be significantly smaller than `1e-30`.